### ISSUE TITLE
[BUG] Cannot build app with version 0.6.0

### ISSUE THREAD

**Issue by m-schmailzl** [juliuscanute/qr_code_scanner#451]
**Describe the bug**
The iOS build fails when using qr_code_scanner version 0.6.0.
```
Command CompileSwiftSources failed with a nonzero exit code
    remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:168:50: error: 'descriptor' is only available in iOS 11.0 or newer
                                        switch (code.descriptor) {
                                                     ^
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:168:50: note: add 'if #available' version check
                                        switch (code.descriptor) {
                                                     ^
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:123:10: note: add @available attribute to enclosing instance method
        func startScan(_ arguments: Array<Int>, _ result: @escaping FlutterResult) {
             ^
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:11:14: note: add @available attribute to enclosing class
    public class QRView:NSObject,FlutterPlatformView {
                 ^
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:169:62: error: 'CIQRCodeDescriptor' is only available in iOS 11.0 or newer
                                        case let qrDescriptor as CIQRCodeDescriptor:
                                                                 ^
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:169:62: note: add 'if #available' version check
                                        case let qrDescriptor as CIQRCodeDescriptor:
                                                                 ^
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:123:10: note: add @available attribute to enclosing instance method
        func startScan(_ arguments: Array<Int>, _ result: @escaping FlutterResult) {
             ^
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:11:14: note: add @available attribute to enclosing class
    public class QRView:NSObject,FlutterPlatformView {
                 ^
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:171:65: error: 'CIAztecCodeDescriptor' is only available in iOS 11.0 or newer
                                        case let aztecDescriptor as CIAztecCodeDescriptor:
                                                                    ^
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:171:65: note: add 'if #available' version check
                                        case let aztecDescriptor as CIAztecCodeDescriptor:
                                                                    ^
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:123:10: note: add @available attribute to enclosing instance method
        func startScan(_ arguments: Array<Int>, _ result: @escaping FlutterResult) {
             ^
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:11:14: note: add @available attribute to enclosing class
    public class QRView:NSObject,FlutterPlatformView {
                 ^
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:173:66: error: 'CIPDF417CodeDescriptor' is only available in iOS 11.0 or newer
                                        case let pdf417Descriptor as CIPDF417CodeDescriptor:
                                                                     ^
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:173:66: note: add 'if #available' version check
                                        case let pdf417Descriptor as CIPDF417CodeDescriptor:
                                                                     ^
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:123:10: note: add @available attribute to enclosing instance method
        func startScan(_ arguments: Array<Int>, _ result: @escaping FlutterResult) {
             ^
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:11:14: note: add @available attribute to enclosing class
    public class QRView:NSObject,FlutterPlatformView {
                 ^
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:175:70: error: 'CIDataMatrixCodeDescriptor' is only available in iOS 11.0 or newer
                                        case let dataMatrixDescriptor as CIDataMatrixCodeDescriptor:
                                                                         ^
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:175:70: note: add 'if #available' version check
                                        case let dataMatrixDescriptor as CIDataMatrixCodeDescriptor:
                                                                         ^
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:123:10: note: add @available attribute to enclosing instance method
        func startScan(_ arguments: Array<Int>, _ result: @escaping FlutterResult) {
             ^
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:11:14: note: add @available attribute to enclosing class
    public class QRView:NSObject,FlutterPlatformView {
                 ^
    /Users/max/flutter/.pub-cache/hosted/pub.dartlang.org/qr_code_scanner-0.6.0/ios/Classes/QRView.swift:193:43: warning: value 'safeResult' was defined but never used; consider replacing with boolean test
                                    guard let safeResult = result else { continue }
                                          ~~~~^~~~~~~~~~~~~
                                                                  != nil
```

**Flutter information**
```
[✓] Flutter (Channel stable, 2.5.3, on macOS 11.6 20G165 darwin-arm, locale de-DE)
    • Flutter version 2.5.3 at /Users/max/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18116933e7 (3 weeks ago), 2021-10-15 10:46:35 -0700
    • Engine revision d3ea636dc5
    • Dart version 2.14.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/max/Library/Android/sdk
    • Platform android-31, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 13.1, Build version 13A1030d
    • CocoaPods version 1.11.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.61.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.27.0

[✓] Connected device (2 available)
    • iPhone 13 (mobile) • 6700ECD7-5DB6-4DDE-9C3E-90AE2D4CB8FD • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-0 (simulator)
    • Chrome (web)       • chrome                               • web-javascript • Google Chrome 95.0.4638.69

• No issues found!
```

**Device (please complete the following information):**
 - Device: iPhone 13 simulator
 - OS: iOS 15.0

**Comment by juliansteenbakker** [comment]
Thank you for reporting, i have found the issue and will release a fix for it asap.

**Comment by juliansteenbakker** [comment]
Released v0.6.1
